orchagent/orchdaemon.cpp

```diff
@@ -589,6 +592,10 @@ bool OrchDaemon::init()
 
     m_orchList.push_back(&CounterCheckOrch::getInstance(m_configDb));
 
+    vector<string> p4rt_tables = {APP_P4RT_TABLE_NAME};
```

Comment:
Should this be done based on global flag? `gResponsePublisherRecord`

Reply:
@baxia-lan @mint570 what do you think?

Reply:
gResponsePublisherRecord is to enable/disable the APPL STATE DB record file. I would suggest to use a different flag to enable/disable p4orch. Brian, is it possible to pass the p4orch enable flag from build time? If not, we can just hardcode one in the code (similar to gResponsePublisherRecord)?

Reply:
We do not need to have a flag here. Initialization of p4orch will not impact anything, it just listens to APP_P4RT_TABLE_NAME("P4RT" table in APP DB). As long as no entry is added in the"P4RT" table in APP DB, P4Orch will do nothing.

Reply:
I think yes, we probably don't need a flag here. Just make sure the p4orch init/constructor is not invoking something that a legacy platform does not support. Like an example below, where we are moving `initAclGroupsBindToSwitch`  to p4orch.

Reply:
It should be safe to call the constructor. `initAclGroupsBindToSwitch` isn't called until a P4 program is pushed.